src/library/trackprocessing.cpp

```diff
@@ -0,0 +1,108 @@
+#include "library/trackprocessing.h"
+
+#include <QProgressDialog>
+#include <QThread>
+
+#include "library/trackcollection.h"
+#include "library/trackcollectionmanager.h"
+#include "util/logger.h"
+
+namespace mixxx {
+
+namespace {
+
+const Logger kLogger("ModalTrackBatchProcessor");
+
+} // anonymous namespace
+
+int ModalTrackBatchProcessor::processTracks(
+        const QString& progressLabelText,
+        TrackCollectionManager* pTrackCollectionManager,
+        TrackPointerIterator* pTrackPointerIterator) {
+    DEBUG_ASSERT(pTrackCollectionManager);
+    DEBUG_ASSERT(pTrackPointerIterator);
+    DEBUG_ASSERT(QThread::currentThread() ==
+            pTrackCollectionManager->thread());
+    int trackCount = 0;
+    int estimatedTotalCount =
+            pTrackPointerIterator->estimateItemsRemaining().value_or(trackCount);
```

Comment:
I am confused. Total count and remaining count are not the same.

Reply:
Estimated = guessed. If unknown we take the number of processed tracks and adjust it when exceeding the maximum.

Reply:
It took me a while of reading the code to understand but it makes sense. Please add a comment:
"Before starting any tasks, total = remaining"

Reply:
Thanks for insisting on a clarification. The update during the loop contains a bug. This never happened, because we only iterate over sequences of known size.